### PR TITLE
Enable Etcd's RBAC

### PR DESCRIPTION
**Pull request by vespian**
## High-level description

This PR:
* adds some linting scripts and configuration in order to make the development easier
* enables RBAC support for DC/OS' etcd. 

From now on it will not be possible to modify etcd contents with any certificate issued by DC/OS' CA, but only the ones with the CN name matching (see [2] for details) one of the users that this PR introduces.

The users that were introduced are as follows:
* `root` - can do anything in etcd, not used normally by the dc/os services, the certificate which gives access to this role is named `etcd-root.crt`
* `adminrouter` - can access all the keys in etcd , both for reading and writing, but cannot make any authn/authz changes. Used by adminrouter to access etcd, once dc/os authentication and authorization grants access. The certificate which gives access to this role is named `adminrouter-grpc-client.crt`
* `calico` - can access all the keys prefixed with `/calico` (see [1] for details), used by all the calico agents while connecting to etcd directly/without adminrouter. The certificate which gives access to this role is named `etcd-client.crt`

[1] https://docs.projectcalico.org/reference/etcd-rbac/calico-etcdv3-paths#caliconode
[2] https://github.com/etcd-io/etcd/blob/467e08c32a6f1ab151f7ab65d2496133289cce47/Documentation/op-guide/authentication.md

## Corresponding DC/OS tickets (required)

  - [D2IQ-59030](https://jira.mesosphere.com/browse/D2IQ-59030) introduce RBAC authn&authz 

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by vespian** [comment]
@mesosphere-mergebot changelog-not-required feature not yet released

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot label ready for review

**Comment by vespian** [comment]
@mesosphere-mergebot request-review  @jkoelker

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@wavesoft @jongiddy Just a friendly reminder - could you please have a look?

**Comment by vespian** [comment]
@jkoelker fooobarrrr

File: packages/etcd/extra/etcd_discovery/etcd_discovery.py [hunk 1]
```diff
@@ -334,9 +337,49 @@ def parse_cmdline() -> argparse.Namespace:
...
+        args.etcd_client_tls_cert,
+        args.etcd_client_tls_key,
+    )
+
+    # See below for more context:
+    # https://github.com/etcd-io/etcd/blob/3898452b5432b4d69028ee79d796ddeab0acc42c/Documentation/op-guide/authentication.md
+    etcdctl.ensure_user("root")
+    etcdctl.grant_role("root", "root")
+
+    etcdctl.ensure_user("calico")
```
Comment (wavesoft) [inline packages/etcd/extra/etcd_discovery/etcd_discovery.py]: I have a feeling that the list of initial users and roles must be placed in a separate file that would be easy for someone to find out and update. But I understand the equally important role of not over-engineering stuff, that leads to accepting the fact that someone else could simply update this file with the new users/roles if needed 😄

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

File: packages/dcos-test-utils/buildinfo.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "single_source" : {
     "kind": "git",
     "git": "https://github.com/dcos/dcos-test-utils.git",
-    "ref": "9a9cafdd509bc9b8c76ab5e9f59849e7f3ace765",
-    "ref_origin": "master"
+    "ref": "aad7e148c66b85787d4217e7763afd5e4835d98a",
+    "ref_origin": "prozlach/D2IQ-59030_etcd_rbac"
```
Comment (jkoelker) [inline packages/dcos-test-utils/buildinfo.json]: Should we wait until that ref is in master before merging this?

Comment (vespian) [reply packages/dcos-test-utils/buildinfo.json]: Definitely! I just wanted  to have something working and ready for review while we are waiting for the issue blocking the update of dcos-test-utils to be resolved (https://jira.d2iq.com/browse/D2IQ-63177)

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot sync

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot sync

**Comment by vespian** [comment]
@mesosphere-mergebot changelog-not-required part of the effort to brin etcd on-board/contained in other changelog entries.

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot sync

**Comment by DominikDary** [comment]
@mesosphere-mergebot sync

**Comment by vespian** [comment]
@mesosphere-mergebot bump-ee

**Comment by vespian** [comment]
@mesosphere-mergebot label  ready for review

**Comment by vespian** [comment]
@mesosphere-mergebot label ready for review

**Comment by vespian** [comment]
@mesosphere-mergebot

**Comment by vespian** [comment]
@mesosphere-mergebot sync

**Pull request merged**